### DISCUSSION TITLE
Possible to filter an array of JSON objects?

### DISCUSSION THREAD

**Discussion by richardkyk** [supabase/supabase#4422] [Questions] [answered]
I have a `Review` table, which links back to an `Item` using a FK `itemId` to allow for a O2M relationship (an item has many reviews).

```
Review
{
  id: 1,
  itemId: 34232,
  rating: 7,
  type: 'primary',
  notes: 'very cool'
}
```

I have a view `all_items` with a JSONB column
```
CREATE OR REPLACE VIEW all_items AS
SELECT
  i.id as "ID",
  i.name as "Name",
  "Reviews"
FROM "Item" i
LEFT JOIN (
  SELECT "itemId",
    jsonb_agg(
      to_jsonb("Review")
    ) as "Reviews"
  FROM "Review"
  GROUP BY "itemId"
) r on i.id = r."itemId"
GROUP BY i.id, "Reviews";
```
which I can query using `all_items?Reviews->0->>type=eq.primary&Reviews->0->>rating=gte.5`
however this only returns items where the **first** review in the array is of type primary and has a rating >= 5.

Is there a way to find all items which have at least one review with a rating >= 5 and is of type primary,
or is there a better way to query/structure the data?

Current workaround I have found is to use
`all_items?not.or=(Reviews.cs.[{"rating": 5}],Reviews.cs.[{"rating": 4}],Reviews.cs.[{"rating": 3}],Reviews.cs.[{"rating": 2}],Reviews.cs.[{"rating": 1}])&Reviews.cs[{"type": "primary"}]`

**Comment by soedirgo** [comment, marked as answer]
This is now possible without needing a view with the new [`!inner`](https://github.com/supabase/postgrest-js/issues/197) filter. Try:
```js
const { data, error } = await supabase
  .from('Item')
  .select('*, Review!inner(*)')
  .eq('Review.type', 'primary')
  .gte('Review.rating', 5)
```
or:
```
Item?select=*,Review!inner(*)&Review.type=eq.primary&Review.rating=gte.5
```

**Reply by richardkyk** [reply]
This is perfect thank you!
